frontend/src/ui/Access.tsx

```diff
@@ -670,6 +765,117 @@ const ActionMenuItem: React.FC<ActionMenuItemProps> = (
+        || inheritedAcl?.get(r)?.actions.has(permissionLevels.highest))
+    );
+
+    const selectionIsInitial = selections.size === initialAcl.size
+        && [...selections].every(([role, info]) => {
+            const other = initialAcl.get(role);
+            return other && areSetsEqual(other.actions, info.actions);
+        });
+
+    const submit = onSubmit;
+
+    useNavBlocker(!selectionIsInitial);
+
+    return (
+        <div {...{ className }} css={{
+            display: "flex",
+            gap: 8,
+            alignSelf: "flex-start",
+            marginTop: 40,
+        }}>
+            {/* Reset button */}
+            <Button
+                disabled={selectionIsInitial}
+                onClick={() => currentRef(resetModalRef).open()}
+                css={{ ...!selectionIsInitial && { ":hover": { color: COLORS.danger0 } } }}
+            >
+                {t("manage.access.reset-modal.label")}
+            </Button>
+            <Modal ref={resetModalRef} title={t("manage.access.reset-modal.title")}>
+                <p>{t("manage.access.reset-modal.body")}</p>
+                <div css={{
+                    display: "flex",
+                    gap: 12,
+                    justifyContent: "center",
+                    flexWrap: "wrap",
+                    marginTop: 32,
+                }}>
+                    <Button onClick={() => currentRef(resetModalRef).close?.()}>
+                        {t("general.action.cancel")}
+                    </Button>
+                    <Button kind="danger" onClick={() => {
+                        setSelections(initialAcl);
+                        currentRef(resetModalRef).close?.();
+                    }}>{t("manage.access.reset-modal.label")}</Button>
+                </div>
+            </Modal>
+
+            {/* Save button */}
+            <Button
+                disabled={selectionIsInitial}
+                onClick={() =>
+                    !containsUser(selections)
+                        ? currentRef(saveModalRef).open()
+                        : submit(selections)
+                }
+                css={{ ...!selectionIsInitial && { ":hover": { color: COLORS.happy0 } } }}
+            >{t("general.action.save")}</Button>
+            {inFlight && <div css={{ marginTop: 16 }}><Spinner size={20} /></div>}
+            <ConfirmationModal
+                ref={saveModalRef}
+                title={t("manage.access.save-modal.title")}
+                buttonContent={t("manage.access.save-modal.confirm")}
+                onSubmit={() => submit(selections)}
+            >
+                {/*
+                    This isn't pretty, but i18next needs strongly typed variables for interpolation
+                    and doesn't allow partials.
+                */}
+                <p>{t(`manage.access.save-modal.disclaimer-${
+                    permissionLevels.highest as "write" | "admin"
+                }`)}</p>
```

Comment:
Can't you rather... have a prop `kind: "write" | "admin"` that you use here? Then you don't have to use the ACL context here, also not needing to make these buttons a child of the AclSelector. The buttons are used in two places and for each, there can be a fixed `"write"` or `"admin"`. I hope it's clear what I mean.

Reply:
Yeah I suppose, but is having access to the ACL context really a bad thing? I mean... nevermind, I think both of us just want to get this over with.